### PR TITLE
Walkthrough coadd updates

### PR DESCRIPTION
**Pull request by Rplesha**
@emsnyder we should coordinate merging our two branches to resolve conflicts better, but first let me know how the changes I made look and if there are any other edits I should make!

### CODE REVIEW

File: notebooks/ullyses-walkthrough.ipynb [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "\n",
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:69]: missing a closing parantheses around ULLYSES in the first paragraph

File: notebooks/ullyses-walkthrough.ipynb [hunk 2]
```diff
@@ -69,7 +69,7 @@
...
+    "The ULLYSES team produces several types of High Level Science Products (HLSPs), which we will walk through in this notebook. Products are made using both archival data and new HST observations obtained through the ULLYSES program. Data products are available from the [ULLYSES search catalog](https://mast.stsci.edu/search/ui/#/ullyses) (HLSPs only), the [MAST Data Discovery Portal](https://mast.stsci.edu/) (HLSPs and contributing data), or directly as a High-Level Science Product collection using the [DOI](https://archive.stsci.edu/hlsp/ullyses) (HLSPs only). More information about ULLYSES data products can be found under [\"Data\"](https://ullyses.stsci.edu/ullyses-download.html) on the ULLYSES website."
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:72]: maybe change to "under the Data tab"

File: notebooks/ullyses-walkthrough.ipynb [hunk 3]
```diff
@@ -109,12 +110,42 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up a quick function to bin the spectra\n",
+    "def downsample_1d(myarr,factor):\n",
+    "    \"\"\"\n",
+    "    Downsample a 1D array by averaging over *factor* pixels.\n",
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:134]: maybe change to "`<factor>` pixels" which is more standard notation

File: notebooks/ullyses-walkthrough.ipynb [hunk 4]
```diff
@@ -156,10 +187,9 @@
     "|--------------------------------------------------------------------------------------------------------------|-----------|---------------|-------------------------------------------------|-------------------|-------|\n",
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:187]: this sentence is a little hard to parse. suggest rephrasing: "The file names for ULLYSES science data products follow a naming scheme which encodes the target designation and the instruments and observing configuration(s) that contribute to the product. However, not all products will appear in the early releases. File names have the form" -> "The file names for ULLYSES science data products follow a specific naming scheme. This includes the target name, instrument(s), and observing configuration(s) that contributed to the product. However, not all products will appear in the early releases. File names have the form"

Comment (jotaylor) [reply notebooks/ullyses-walkthrough.ipynb:187]: Also missing a comma between the `<target>` and `<version>` comment

File: notebooks/ullyses-walkthrough.ipynb [hunk 5]
```diff
@@ -261,27 +316,43 @@
...
     "\\\n",
-    "From the filename, we can see:\n",
+    "From the filename, `hlsp_ullyses_hst_cos-stis_v-xx-cha_uv-opt_dr7_preview-spec.fits`, we can see:\n",
     "- the telescope used is HST\n",
-    "- the instrument used are COS and STIS\n",
-    "- the optical element is UV (in this case, many optical elements made up this data product)\n",
-    "- this file was created for data release 6\n",
+    "- the instruments used are COS and STIS\n",
+    "- the target observed is V-XX-CHA\n",
+    "- the optical range is UV and Optical (in this case, many optical elements made up this data product)\n",
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:347]: I am not 100% sure what this means, can you explain further? "(in this case, many optical elements made up this data product)"

File: notebooks/ullyses-walkthrough.ipynb [hunk 6]
```diff
@@ -261,27 +316,43 @@
...
+    "- this file was created for data release 7\n",
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:348]: should "data release 7" be capitalized? not sure haha

File: notebooks/ullyses-walkthrough.ipynb (hunk 5 above)
Comment (jotaylor) [reply notebooks/ullyses-walkthrough.ipynb:347]: Also I think this (and the target below it) should say "wavelength range" rather than "optical range"

File: notebooks/ullyses-walkthrough.ipynb [hunk 7]
```diff
@@ -302,7 +373,17 @@
...
+    "print(spec_filename)\n",
+    "with fits.open(spec_filename) as spec_hdu:\n",
+    "    print(spec_hdu[1].columns)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's plot the flux, error, wavelength, and SNR for each level 4 (`preview-spec`) dataset. To get these values we use \".data\" to access the data in the science extension of the HDU. To access the header of this extension instead, we would use \".header\""
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:386]: Suggest putting .data and .header in code block quotes like `.data` and `.header`

File: notebooks/ullyses-walkthrough.ipynb [hunk 8]
```diff
@@ -311,58 +392,228 @@
...
+    "    axs[2].set_ylabel('SNR')\n",
+    "    plt.xlabel('Wavelength [$\\mathrm{\\AA}$]')"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "#### Single-epoch Provenance Table\n",
+    "Now that we know what the level 4 product (`preview-spec`) looks like, let's see how the exposure level combined level 2 products (`cspec`) compare."
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:456]: Suggest dropping "exposure level" or using a different description.

File: notebooks/ullyses-walkthrough.ipynb [hunk 9]
```diff
@@ -311,58 +392,228 @@
...
+    "    else:\n",
+    "        # set the limit based on the preview-spec to avoid grating edges\n",
+    "        axs[0].set_ylim(np.nanmin(flux), np.nanmax(flux)) \n",
+    "        axs[1].set_ylim(np.nanmin(error), np.nanmax(error)) \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:534]: It might help to add some commentary on the plots here.

File: notebooks/ullyses-walkthrough.ipynb [hunk 10]
```diff
@@ -311,58 +392,228 @@
...
+    "#### Non-Variable Provenance Table\n",
     "\n",
-    "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table. The fields in the table are metadata harvested from the headers of the contributing spectra."
+    "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table. The fields in the table are metadata harvested from the headers of the contributing spectra. Notice that we already used the \"DISPERSER\" column in the plot labels above to see all contributing gratings in that product."
```
Comment (jotaylor) [inline notebooks/ullyses-walkthrough.ipynb:616]: May want to reword the first sentence a bit. Maybe something like: "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table." -> "For each spectrum that contributes to a given HLSP, select metadata is available in the provenance table."

**Review by jotaylor — CHANGES_REQUESTED** [review]
Some minor grammatical comments, feel free to incorporate those you agree with.

File: notebooks/ullyses-walkthrough.ipynb (hunk 5 above)
Comment (Rplesha) [reply notebooks/ullyses-walkthrough.ipynb:347]: @jotaylor I think this was left over from the old naming scheme for the preview spec files for DR6 where the gratings were listed instead of "uv" or "opt". I've just removed that comment since it's a little confusing now.

File: notebooks/ullyses-walkthrough.ipynb (hunk 6 above)
Comment (Rplesha) [reply notebooks/ullyses-walkthrough.ipynb:348]: @emsnyder what do you think? I think it depends if we want to say if it's a proper noun or not, but I don't feel strongly either way.

File: notebooks/ullyses-walkthrough.ipynb (hunk 9 above)
Comment (Rplesha) [reply notebooks/ullyses-walkthrough.ipynb:532]: @jotaylor I've added some text below the cspec and aspec plots. Let me know if they sound ok!

Comment (emsnyder) [reply notebooks/ullyses-walkthrough.ipynb:532]: These sound good to me

File: notebooks/ullyses-walkthrough.ipynb (hunk 6 above)
Comment (emsnyder) [reply notebooks/ullyses-walkthrough.ipynb:346]: I'm indifferent. It does sound like a proper noun, but also think it's okay for now

**Review by emsnyder — APPROVED** [review]
LGTM!

File: notebooks/ullyses-walkthrough.ipynb (hunk 9 above)
Comment (jotaylor) [reply notebooks/ullyses-walkthrough.ipynb:532]: I fixed a typo in one and replaced this sentence 
"so the more indepth notebook shows how to use the ULLYSES coadd and abut code to create your own products"
with "For this reason, the ULLYSES team recommends using the individual `cspec` files for scientific purposes.". We don't actually support custom abutment right now 😬

**Review by jotaylor — APPROVED** [review]
